CMakeLists.txt

```diff
@@ -38,72 +39,69 @@ if(TILEDMM_GPU_BACKEND STREQUAL "CUDA")
-    endif()
-    if(ROCBLAS_HAS_CGEMM)
-        list(APPEND TILEDMM_DEFINITIONS TILED_MM_ROCBLAS_HAS_CGEMM)
-    endif()
-    if(ROCBLAS_HAS_ZGEMM)
-        list(APPEND TILEDMM_DEFINITIONS TILED_MM_ROCBLAS_HAS_ZGEMM)
+    find_package(rocblas REQUIRED)
+    if (NOT TARGET tiledmm::rocm)
+      add_library(tiledmm::rocm INTERFACE IMPORTED)
+      target_link_libraries(tiledmm::rocm INTERFACE roc::rocblas)
     endif()
+
+    list(APPEND TILEDMM_DEFINITIONS TILED_MM_ROCM)
+    # TODO: it should be possible to remove this (and adapt the cpp code
+    # accordingly). Recent rocblas supports all types
+    list(APPEND TILEDMM_DEFINITIONS TILED_MM_ROCBLAS_HAS_SGEMM
+      TILED_MM_ROCBLAS_HAS_DGEMM TILED_MM_ROCBLAS_HAS_ZGEMM)
 else()
     message(FATAL_ERROR "TILED_MM: GPU Backend must be CUDA or ROCM!")
 endif()
 
 # Tiled-MM
 #
-add_subdirectory(src/Tiled-MM)
+include(CMakePackageConfigHelpers)
+include(GNUInstallDirs)
 
-if(TILEDMM_WITH_INSTALL)
-    include(CMakePackageConfigHelpers)
-    include(GNUInstallDirs)
+add_subdirectory(src/Tiled-MM)
 
-    install(EXPORT Tiled-MM_targets
-            FILE Tiled-MMTargets.cmake
-            NAMESPACE Tiled-MM::
-            DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/Tiled-MM")
+install(EXPORT Tiled-MM_targets
+  FILE Tiled-MMTargets.cmake
+  NAMESPACE Tiled-MM::
+  DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/Tiled-MM")
 
-    install(DIRECTORY "${Tiled-MM_SOURCE_DIR}/src/Tiled-MM"
-            DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}"
-            FILES_MATCHING PATTERN "*.hpp")
+install(DIRECTORY "${Tiled-MM_SOURCE_DIR}/src/Tiled-MM"
+  DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}"
+  FILES_MATCHING PATTERN "*.hpp")
 
-    write_basic_package_version_file("${Tiled-MM_BINARY_DIR}/Tiled-MMConfigVersion.cmake"
-                                     VERSION ${Tiled-MM_VERSION}
-                                     COMPATIBILITY SameMajorVersion)
+write_basic_package_version_file("${Tiled-MM_BINARY_DIR}/Tiled-MMConfigVersion.cmake"
+  VERSION ${Tiled-MM_VERSION}
+  COMPATIBILITY SameMajorVersion)
 
-    configure_file("${Tiled-MM_SOURCE_DIR}/cmake/Tiled-MMConfig.cmake.in"
-                   "${Tiled-MM_BINARY_DIR}/Tiled-MMConfig.cmake"
-                   @ONLY)
+configure_file("${Tiled-MM_SOURCE_DIR}/cmake/Tiled-MMConfig.cmake.in"
+  "${Tiled-MM_BINARY_DIR}/Tiled-MMConfig.cmake"
+  @ONLY)
 
-    install(FILES "${Tiled-MM_BINARY_DIR}/Tiled-MMConfig.cmake"
-                  "${Tiled-MM_BINARY_DIR}/Tiled-MMConfigVersion.cmake"
-                  "${Tiled-MM_SOURCE_DIR}/cmake/FindROCBLAS.cmake"
-                  "${Tiled-MM_SOURCE_DIR}/cmake/FindHIPLIBS.cmake"
-            DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/Tiled-MM")
-endif()
+install(FILES "${Tiled-MM_BINARY_DIR}/Tiled-MMConfig.cmake"
+  "${Tiled-MM_BINARY_DIR}/Tiled-MMConfigVersion.cmake"
+  DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/Tiled-MM")
 
 if(TILEDMM_WITH_TESTS OR TILEDMM_WITH_EXAMPLES)
+  include(GitSubmodule)
```

Comment:
we should probably remove cxxopts as submodule as we are not developing it. Treating this as external dependency is better.

Reply:
Oh, I forgot to remove it, it's also inconsistent with COSMA where cxxopts is a external-only dependency.

Reply:
no worries